pkg/inject/inject.go

```diff
@@ -202,16 +203,10 @@ func (conf *ResourceConfig) GetOwnerRef() *metav1.OwnerReference {
 // from the namespace they belong to. If the namespace has a valid config key
 // that the pod does not, then it is appended to the pod's template
 func (conf *ResourceConfig) AppendNamespaceAnnotations() {
-	for _, key := range ProxyAnnotations {
-		if _, found := conf.nsAnnotations[key]; !found {
-			continue
-		}
-		if val, ok := conf.GetConfigAnnotation(key); ok {
-			conf.AppendPodAnnotation(key, val)
-		}
-	}
+	annotations := append(ProxyAnnotations, ProxyAlphaConfigAnnotations...)
+	annotations = append(annotations, k8s.ProxyInjectAnnotation)
 
-	for _, key := range ProxyAlphaConfigAnnotations {
+	for _, key := range annotations {
```

Comment:
this is just cleanup and not really related to the fix, right?

Reply:
yeah, to avoid repeating for a third time (for the inject annotation) the same checks